---
layout: default
title: History
---
# History of the Needleman-Wunsch Algorithm

   The Needleman-Wunsch algorithm is an algorithm developed to align protein or
nucleotide sequences. This algorithm solves the problem of global alignment. Global alignment
is the aligning of two entire sequences compared to local alignment which is finding parts of the
two sequences that are most similar. This is one of the first uses of dynamic programming to
compare two DNA sequences. The Needleman-Wunsch algorithm was created by Saul B.
Needleman and Christian D. Wunsch in 1970.

  The original purpose of this algorithm was to find similarities between amino acid
sequences of two proteins. The original algorithm was to penalize the alignment only for
matches and mismatches, and no penalty for gaps. Today, most implementations penalize the
score of the alignment for gaps too.

  Needleman and Wunsch developed this algorithm with the idea of maximizing
similarities between two sequences. Later other Bioinformatics algorithms were developed to
minimize differences. The Needleman-Wunsch algorithm is still commonly used today for global
alignment of two sequences, especially when correctness if most important.

[Head back to the Main Page](https://jsebcort.github.io/NeedlemanWunsch/)
